test/unit/Github/Api/V3/CreateMilestoneTest.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\AutomaticReleases\Test\Unit\Github\Api\V3;
+
+use Laminas\AutomaticReleases\Git\Value\SemVerVersion;
+use Laminas\AutomaticReleases\Github\Api\V3\CreateMilestoneFailed;
+use Laminas\AutomaticReleases\Github\Api\V3\CreateMilestoneThroughApiCall;
+use Laminas\AutomaticReleases\Github\Value\RepositoryName;
+use Laminas\Diactoros\Request;
+use Laminas\Diactoros\Response;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Log\LoggerInterface;
+use Webmozart\Assert\Assert;
+
+use function uniqid;
+
+class CreateMilestoneTest extends TestCase
+{
+    /** @var ClientInterface&MockObject */
+    private $httpClient;
+    /** @var RequestFactoryInterface&MockObject */
+    private $messageFactory;
+    /** @var LoggerInterface&MockObject */
+    private $logger;
+    /** @psalm-var non-empty-string */
+    private string $apiToken;
+    private CreateMilestoneThroughApiCall $createMilestone;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->httpClient     = $this->createMock(ClientInterface::class);
+        $this->messageFactory = $this->createMock(RequestFactoryInterface::class);
+        $this->logger         = $this->createMock(LoggerInterface::class);
+        $apiToken             = uniqid('apiToken', true);
+
+        Assert::notEmpty($apiToken);
+
+        $this->apiToken        = $apiToken;
+        $this->createMilestone = new CreateMilestoneThroughApiCall(
+            $this->messageFactory,
+            $this->httpClient,
+            $this->apiToken,
+            $this->logger
+        );
+    }
+
+    public function testSuccessfulRequest(): void
+    {
+        $this
+            ->messageFactory
+            ->expects(self::any())
+            ->method('createRequest')
+            ->with('POST', 'https://api.github.com/repos/foo/bar/milestones')
+            ->willReturn(new Request('https://the-domain.com/the-path'));
+
+        $validResponse = (new Response())->withStatus(201);
+
+        $validResponse->getBody()->write(<<<'JSON'
+{
+    "html_url": "http://another-domain.com/the-pr"
+}
+JSON
```

Comment:
Indent these, please; you can see in some of the other tests (e.g., the keep-a-changelog related tests) where I've done that. Makes reading easier, and supported since PHP 7.3. You just need to have the end marker indented the same amount as the rest of the lines.